File: killrvideo-service-videocatalog/src/main/java/com/killrvideo/service/video/dao/VideoCatalogMapper.java
package com.killrvideo.service.video.dao;

import com.datastax.oss.driver.api.core.CqlSession;
import com.datastax.oss.driver.api.mapper.MapperBuilder;
import com.datastax.oss.driver.api.mapper.annotations.DaoFactory;
import com.datastax.oss.driver.api.mapper.annotations.Mapper;

@Mapper
public interface VideoCatalogMapper {
    @DaoFactory
    VideoDao getVideoDao();

    @DaoFactory
    LatestVideoDao getLatestVideoDao();

    @DaoFactory
    UserVideoDao getUserVideoDao();

    static MapperBuilder<VideoCatalogMapper> build(CqlSession session) {
        return new VideoCatalogMapperBuilder(session);
    }
}

File: killrvideo-service-videocatalog/src/main/kotlin/com/killrvideo/service/video/dao/VideoCatalogMapper.kt
package com.killrvideo.service.video.dao

import com.datastax.oss.driver.api.mapper.annotations.DaoFactory
import com.datastax.oss.driver.api.mapper.annotations.Mapper

@Mapper
interface VideoCatalogMapper {
    @get:DaoFactory
    val videoDao: VideoDao

    @get:DaoFactory
    val latestVideoDao: LatestVideoDao

    @get:DaoFactory
    val userVideoDao: UserVideoDao
}
